Src/AutoFixture/RandomNumericSequenceGenerator.cs

```diff
@@ -104,54 +105,18 @@ private static void ValidateThatLimitsAreStrictlyAscending(long[] limits)
 
         private object CreateRandom(Type request)
         {
-            switch (Type.GetTypeCode(request))
+            if(!request.IsNumeric())
             {
-                case TypeCode.Byte:
-                    return (byte)
-                        this.GetNextRandom();
-
-                case TypeCode.Decimal:
-                    return (decimal)
-                        this.GetNextRandom();
-
-                case TypeCode.Double:
-                    return (double)
-                        this.GetNextRandom();
-
-                case TypeCode.Int16:
-                    return (short)
-                        this.GetNextRandom();
-
-                case TypeCode.Int32:
-                    return (int)
-                        this.GetNextRandom();
-
-                case TypeCode.Int64:
-                    return
-                        this.GetNextRandom();
-
-                case TypeCode.SByte:
-                    return (sbyte)
-                        this.GetNextRandom();
-
-                case TypeCode.Single:
-                    return (float)
-                        this.GetNextRandom();
-
-                case TypeCode.UInt16:
-                    return (ushort)
-                        this.GetNextRandom();
-
-                case TypeCode.UInt32:
-                    return (uint)
-                        this.GetNextRandom();
-
-                case TypeCode.UInt64:
-                    return (ulong)
-                        this.GetNextRandom();
-
-                default:
-                    return new NoSpecimen();
+                return new NoSpecimen();
+            }
+            var nextRandom = GetNextRandom();
+            try
+            {
+                return Convert.ChangeType(nextRandom, request, CultureInfo.InvariantCulture);
+            }
+            catch(OverflowException)
+            {
+                return Convert.ChangeType(0, request, CultureInfo.InvariantCulture);

```

Comment:
See [the above comment](#r51185435).

The only difference is that, for reasons I haven't investigated, it looks like the maximum is reached sooner than 256 iterations:

```F#
> let rnsg = RandomNumericSequenceGenerator ();;
val rnsg : RandomNumericSequenceGenerator

> [1..300] |> List.map (fun _ -> rnsg.Create(typeof<byte>, null)) |> List.skip 100;;
val it : obj list =
  [43uy; 83uy; 29uy; 94uy; 5uy; 18uy; 65uy; 91uy; 88uy; 22uy; 61uy; 31uy; 7uy;
   55uy; 67uy; 97uy; 75uy; 36uy; 76uy; 89uy; 50uy; 92uy; 53uy; 26uy; 44uy;
   101uy; 45uy; 0uy; 0uy; 0uy; 0uy; 0uy; 0uy; 0uy; 0uy; 0uy; 0uy; 0uy; 0uy;
   0uy; 0uy; 0uy; 0uy; 0uy; 0uy; 0uy; 0uy; 0uy; 0uy; 0uy; 0uy; 0uy; 0uy; 0uy;
   0uy; 0uy; 0uy; 0uy; 0uy; 0uy; 0uy; 0uy; 0uy; 0uy; 0uy; 0uy; 0uy; 0uy; 0uy;
   0uy; 0uy; 0uy; 0uy; 0uy; 0uy; 0uy; 0uy; 0uy; 0uy; 0uy; 0uy; 0uy; 0uy; 0uy;
   0uy; 0uy; 0uy; 0uy; 0uy; 0uy; 0uy; 0uy; 0uy; 0uy; 0uy; 0uy; 0uy; 0uy; 0uy;
   0uy; ...]
```